addons/purchase_stock/tests/test_old_rules.py

```diff
@@ -276,9 +276,10 @@ def test_02_product_route_level_delays(self):
         self.assertEqual(incoming_shipment2.date_deadline, incoming_shipment2_date)
         old_deadline2 = incoming_shipment2.date_deadline
 
-        # Modify the date_planned of the purchase -> propagate the deadline
+        # Modify the date_planned of the purchase -> propagate the scheduled date in a receipt, so the deadline remains unchanged
         purchase_form = Form(purchase)
         purchase_form.date_planned = purchase.date_planned + timedelta(days=1)
         purchase_form.save()
-        self.assertEqual(incoming_shipment2.date_deadline, old_deadline2 + timedelta(days=1), 'Deadline should be propagate')
-        self.assertEqual(incoming_shipment1.date_deadline, old_deadline1 + timedelta(days=1), 'Deadline should be propagate')
+        self.assertEqual(incoming_shipment2.date_deadline, old_deadline2, 'Deadline should remain unchanged')
+        self.assertEqual(incoming_shipment1.date_deadline, old_deadline1, 'Deadline should remain unchanged')
+        self.assertEqual(purchase.picking_ids.scheduled_date, purchase.date_planned, 'Scheduled Date should propagate')
```

Comment:
```suggestion
        self.assertEqual(purchase.order_line.move_ids.scheduled_date, purchase.date_planned, 'Scheduled Date should propagate')
```
I wouldn't rely on this, as it's kind of a side-effect that `picking_ids` doesn't contain the two other steps. I'd rather target the `IN` picking directly to avoid issues in the future.

Reply:
wouldn't that possibly cause a traceback? given that `move_ids` is `one2many`

Reply:
> It would only throw the traceback if there are more than 1 move_id records, and if there are in this test, then you should test the date of each one.

In this test, no its only one move, I'll just add it then but it will be `purchase.order_line.move_ids.picking_id.scheduled_date`